src/zeek.bif

```diff
@@ -5276,7 +5276,7 @@ function from_json%(s: string, t: any, key_func: string_mapper &default=from_jso
 	else
 		{
 		rval->Assign(valid_idx, false);
-		zeek::emit_builtin_error(res.error().c_str());
+                rval->Assign(result_type->FieldOffset("err"), res.error().c_str());
```

Comment:
I just noticed that `from_json()` is unusual in its use of `FieldOffset()`, which loops over the fields until it finds a matching name. Our other BiFs mostly use `n++` style indexing directly with `Assign()`, and since I can pretty easily see `from_record()` in hot codepaths I suggest moving the whole function to that approach (i.e. also above in lines 5249+). We can do that in a follow-up PR.